### ISSUE TITLE
Why do you not maintain these popular projects anymore?

### ISSUE THREAD

**Issue by IonicaBizau** [harthur/brain#72]
@harthur  I was looking at your profile and saw this:

![image](https://cloud.githubusercontent.com/assets/2864371/10456396/cdc64464-71c9-11e5-9245-c117607cc40f.png)

Why do you not maintain these projects anymore? If you don't have time to do that, why do you not search somebody to keep them alive?

**Comment by nickpoorman** [comment]
@harthur - If you would like to move this repo into an organization, I would be happy to maintain it.

**Comment by IonicaBizau** [comment]
@nickpoorman I was going to suggest that. :smile: :+1:

**Comment by FranzSkuffka** [comment]
That would be great. I am using this software in a project thesis because it is just the simplest solution out here :)

**Comment by ilanbiala** [comment]
@harthur any response?

**Comment by IonicaBizau** [comment]
:disappointed: 

> ![image](https://cloud.githubusercontent.com/assets/2864371/10793727/d0117220-7d9a-11e5-9297-c4cf2eae2aab.png)

**Comment by FranzSkuffka** [comment]
how about forking the repos and creating an organisation for maintenance during her absence?

**Comment by jobsamuel** [comment]
I like that idea @nickpoorman :+1:

**Comment by IonicaBizau** [comment]
@FranzSkuffka I am *not* for forking and maintaining. It's much harder to maintain your own fork because most of the people will find this popular repository.

I also vote for moving these projects in an organization, like @nickpoorman suggested. :dizzy:

**Comment by irony** [comment]
I also agree, it would be nice to get an response from @harthur

**Comment by irony** [comment]
She have switched jobs to something secret so maybe she isn't even allowed to reply here: https://harthur.wordpress.com

**Comment by IonicaBizau** [comment]
@irony Oh, interesting read! :book:

**Comment by DinerIsmail** [comment]
Any update on this, guys? It would be awesome if this would be moved into an organization.

**Comment by IonicaBizau** [comment]
I just emailed @harthur! Hope she will reply me and save these nice goodies. :grin: :gift: :pray:

**Comment by ghpabs** [comment]
Thanks @IonicaBizau let us know how it goes.

**Comment by scrussell24** [comment]
I would also like this!

**Comment by michaelpittino** [comment]
Any update on this?

**Comment by IonicaBizau** [comment]
No response from the author *yet*... :disappointed:

**Comment by ghpabs** [comment]
There's an opportunity here for someone to take this over. Preferably someone who understands the codebase and is willing to move the project forward.

**Comment by michaelpittino** [comment]
Yes but as @IonicaBizau said that's not easy to fork and maintain because of popularity of THIS repository.

**Comment by ghpabs** [comment]
Fair point - I agree that moving *this* project to an org would be the best solution - it's a shame the owner isn't responding the messages. :disappointed:

**Comment by IonicaBizau** [comment]
@michaelpittino I contacted GitHub support–maybe they will be able to help us since @harthur had a talk at Passion Projects. Hopefully they will have a solution. :pray:

**Comment by IonicaBizau** [comment]
From GitHub support:

> Unfortunately we don't have any contact details for Heather Arthur from her Passion Projects appearance. Hopefully she'll reply to your email if she's able to help.

**Comment by ghpabs** [comment]
Let's start a petition at change.org. :-)

**Comment by FranzSkuffka** [comment]
yeah, I think 50 people will sign, but I'm down.

**Comment by voxpelli** [comment]
Why not just create a fork, move development there, keep an eye on this repo and point people to the fork whenever they open issues or PR:s here and then if/when the owner of this repo gets back to you then this repo can be updated to point to that repo.

**Comment by FranzSkuffka** [comment]
+1

**Comment by DinerIsmail** [comment]
I also agree with @voxpelli

**Comment by robertleeplummerjr** [comment]
I did contact @harthur about being a maintainer, no reply.  I did read her website, and why she ended up doing so, and totally respect it.  I would really like to continue in her absence, hoping the best for her.  There is a risk that she doesn't like any of the changes we do, but there is also the chance we may end up improving these fantastic libraries.  That being said, there are enough of us and these libraries are useful enough I went ahead and created: https://github.com/harthur-org and forked all libraries.  If you'd like access, open a pull request.

**Comment by IonicaBizau** [comment]
@robertleeplummerjr That's cool! :+1: 

I would still recommend having real source repositories, not forks (that means: delete forks, create repos, `git push --all` from local clones of @harthur's repos). I do not like forks because they have limitations like:

 - they are not searchable
 - commits in forks are not counted in the contributions calendar (big thing, right? :joy:)

Instead we can explicitly state it's the repos are a continuation of the original projects.

**Comment by FranzSkuffka** [comment]
I agree with @IonicaBizau, that makes perfect sense.

**Comment by robertleeplummerjr** [comment]
@ilanbiala, What if @harthur decides to re-maintain?  We open a single pull request, and merge, and then the org can go away.

This just happened: https://github.com/harthur-org/brain/commit/1668939d7f73c22afa85d0957a189045ad6e940c

**Comment by IonicaBizau** [comment]
@robertleeplummerjr Note we can always make pull requests between source repositories (not only between forks)—I still think we should convert them in source repositories, not forking. :grin:

**Comment by robertleeplummerjr** [comment]
Give me a few, I'll work it out.

**Comment by IonicaBizau** [comment]
> Note we can always make pull requests between source repositories

Well, actually that's wrong: I mean, we can synchronize the source repositories anyways: when the original owner is back and wants to maintain the source repo, we fork it, synchronize the fork (`git pull ...`) and then creating a PR on GitHub. :grin:

**Comment by robertleeplummerjr** [comment]
I get what you intended to say.  I do it all the time for private repos based off of public ones.  Don't know why I didn't think about it, probably an A.D.D. moment.

**Comment by robertleeplummerjr** [comment]
Done for brain, and: https://github.com/harthur-org/brain/commit/388e5ea1840a0659156d560f3bb411acdf1ed535

**Comment by IonicaBizau** [comment]
@robertleeplummerjr :cool: We now can start merging these PRs from this very repository in the @harthur-org account (by `git pull`ing). :sparkles:

**Comment by robertleeplummerjr** [comment]
@IonicaBizau I'd like to make you a fellow maintainer since you started this issue.  Seems only right.  Just pushed unit tests for the `likely` method.

**Comment by IonicaBizau** [comment]
@robertleeplummerjr Thanks! :+1: Happy to help in my spare time. Not sure, but I may use this module in one of projects. :smile:

**Comment by robertleeplummerjr** [comment]
:fist: :fire:

**Comment by robertleeplummerjr** [comment]
Added you to the "with great power comes great responsibility" team.

**Comment by IonicaBizau** [comment]
@robertleeplummerjr Thanks! I'm not evil. :four_leaf_clover:

**Comment by robertleeplummerjr** [comment]
I know, the name of the team is a reminder for me :stuck_out_tongue:

**Comment by IonicaBizau** [comment]
:joy:

**Comment by nickpoorman** [comment]
I love that the community has started maintaining this project. One reason this project is so useful to people, is the simplicity of the code. There's no underlying BLAS library being called to do "magic" behind the scenes. In the spirt of this, I hope when these pull requests start being merged you keep that in mind, and document any new features added. Such as the "likely", which in my opinion should be an a separate util lib and not in the `lib/neuralnetwork.js` source (especially since convolutional neural networks are better suited for that task).

**Comment by IonicaBizau** [comment]
@nickpoorman I agree! :+1: Maybe for such things, open issues in the new repository. It's easier to track. :smile:

**Comment by robertleeplummerjr** [comment]
Well stated @nickpoorman.

**Comment by robertleeplummerjr** [comment]
How would you guys feel about a :+1: policy?  Where at least _one person that has contributed_ and that is _not the committer_ must :+1: the PR moving forward.  When the :+1: is landed, we then can merge the PR.

**Comment by FranzSkuffka** [comment]
Obviously harthur is the maintainer of the npm packages aswell. I have had experience with the npm removing coffeescript (which was not coffee-script) upon my request, because it was a confusing, unnecessarily publicated package.

We could either a.) ask at npmjs.org for permission to be added as package maintainers or b.) publish the packages with a prefix like `harthur-org-brain`.

**Comment by FranzSkuffka** [comment]
@robertleeplummerjr I agree with you on establishing a [consensus](https://en.wikipedia.org/wiki/Participative_decision-making#Consensus)-based policy. However I would extend the pull-request cycle by 2 days for example after the :+1:. During these two days anyone can throw in a veto, delay the merge by another day when a valid argument was given.

**Comment by IonicaBizau** [comment]
@FranzSkuffka I always had good experiences with the guys from npm support—they transferred quite few packages on my account (mostly they were unmaintained/obsolete/bad quality/etc packages), but always they did a great job. I'd say we can give a try to ask them. Then we will get some traffic from the npm package. :joy:

**Comment by FranzSkuffka** [comment]
@IonicaBizau agreed, would be great to get some traffic from there!

btw, WTF did you do to your commit graph :laughing:

**Comment by robertleeplummerjr** [comment]
when one has the capability to rewrite git history...

**Comment by IonicaBizau** [comment]
@FranzSkuffka Check [this](https://github.com/IonicaBizau/github-contributions) out. :joy: 

<sub>Pleeeease, without the *f* word!</sub>

**Comment by IonicaBizau** [comment]
>  ask at npmjs.org for permission to be added as package maintainers

I contacted npm support and they did talk to @harthur. She wants to keep the npm packages in this state, so, we need to apply plan b): to publish the packages under different names. :disappointed: 

/cc @FranzSkuffka

**Comment by ghpabs** [comment]
Ok. Let's think of a good prefix for the package names.

Sample:

`contrib-brain`
`community-brain`
`com-brain`

...

**Comment by silentrob** [comment]
Ill throw out:

`js-brain`
`org-brain`

**Comment by AgentME** [comment]
Total bike-shedding here, but I feel obligated to mention a few common fork naming conventions:

`brain-ng`
`brain-plus`

**Comment by ghpabs** [comment]
Cool - I didn't know there was a naming convention for it.

Where is the easiest place to start an online poll?

**Comment by SoullessWaffle** [comment]
@ghpabs You could try http://strawpoll.me/

**Comment by IonicaBizau** [comment]
My ideas are:

 - `brain.js`
 - `brain2`

:grin:

**Comment by ghpabs** [comment]
Thanks @SoullessWaffle.

[See poll here.](http://strawpoll.me/7307773)

Ps. It's okay if you guys decide to add more options to the poll though.

**Comment by ghpabs** [comment]
#### Results <sup>as of Apr 8, 2016, 10:43 AM GMT+1</sup>

1. brain.js **10 votes** (45%)
2. js-brain **4 votes** (18%)
3. brain-plus **2 votes** (9%)
4. brain2 **2 votes** (9%)
5. contrib-brain **1 vote** (5%)
6. community-brain **1 vote** (5%)
7. org-brain **1 vote** (5%)
8. brain-ng **1 vote** (5%)
9. com-brain **0 votes** (0%)

**Comment by SoullessWaffle** [comment]
Here's two more ideas:
brain-next
brain++ (as in c++; brain incremented by one)

**Comment by IonicaBizau** [comment]
@SoullessWaffle I don't think `+` can be used in the package names.

**Comment by FranzSkuffka** [comment]
`harthur-brain` could be a good choice SEO-wise

**Comment by SoullessWaffle** [comment]
@IonicaBizau you're right, perhaps we could use `brainpp`

**Comment by hadesara** [comment]
Can we decide on the date by which everyone should vote.

**Comment by ghpabs** [comment]
The poll has been open for just over a week now.

Updated results:

1. brain.js 13 votes (46%)
2. js-brain 4 votes (14%)
3. brain2 4 votes (14%)
4. community-brain 2 votes (7%)
5. brain-plus 2 votes (7%)
6. contrib-brain 1 vote (4%)
7. org-brain 1 vote (4%)
8. brain-ng 1 vote (4%)
9. com-brain 0 votes (0%)

28 total votes

@IonicaBizau I guess you could call `brain.js` the winner?

**Comment by IonicaBizau** [comment]
I `npm publish`ed it under the `brain.js` name on npm. :rocket: 

If any of you wants to have superpowers on npm, feel free to :bell: me and I will `npm owner add` you. :grin:  Also, if you want to contribute to the project we can add you as member in the @harthur-org organization.

/cc https://github.com/harthur-org/brain.js/issues/4

**Comment by FranzSkuffka** [comment]
Thank you @IonicaBizau

**Comment by robertleeplummerjr** [comment]
Devs, I believe a "huzza" is in order.

**Comment by robertleeplummerjr** [comment]
Huzza!!!!

**Comment by roccomuso** [comment]
Great job guys. ;)

**Comment by antoniodeluca** [comment]
In the past I have used Brain.js and some other JavaScript based machine learning libraries but unfortunately I have found them not matching my needs. That's why, working on my personal projects, I have developed the idea of creating a different approach by myself. So, sorry to be here to talk about another project but I would really like to receive some opinions and suggestions from experienced people that, being into this specific field, could help me to set useful and shared expectations. By the way, thanks for the great support given to Brain.js so far.

**Comment by robertleeplummerjr** [comment]
Can you please open a new issue, that way this issue stays focused?

**Comment by shimondoodkin** [comment]
https://medium.com/@heatherarthur/what-happens-when-you-retire-from-github-979fb6b143b3

Basically, he writes: This is hard work for him, A one with limited resources to maintain his repositories for others. He is overwhelmed by a large number of people contacting him. He has no resources for this. He dislikes this. So he quit GitHub. he  "declared bankruptcy" on time devoted to Github, and He regrets sharing his repositories because it took so much time from his life. He expected people would fork and the code would be maintained by users themselves. but many good wishing people fall into his open door smashing him because they are just oo much for him.

**Comment by robertleeplummerjr** [comment]
>expected people would fork and the code would be maintained by users themselves

And that is _exactly_ what happened.

**Comment by kjprice** [comment]
This is one of the most fascinating issues I have ever scrolled through. I love coders and the code community. Way to go for pulling together, people!

**Comment by k1r0s** [comment]
what happened to @harthur ? lol

**Comment by janwirth** [comment]
She's alive and well - machine-learning-engineering. I think there was more than one issue motivating her to abandon github. But I don't want to seed any rumors here. You better google any information up yourself. [She has a medium account](https://medium.com/@heatherarthur).

**Comment by nclsjstnn** [comment]
https://medium.com/@heatherarthur/what-happens-when-you-retire-from-github-979fb6b143b3

**Comment by vickylance** [comment]
@FranzSkuffka But her profile says *deceased* https://github.com/harthur what happened to her. It saddens me to see a github profile like that.

**Comment by janwirth** [comment]
Wow - that must be very recent. The last post on medium was Feb 24.

edit: She's alive. Or Twitter has an API for ghosts.
https://twitter.com/harthvader

**Comment by youngkaneda** [comment]
I'm not still understand why she left the Github, someone can explain?

**Comment by AlibekJ** [comment]
Why do you care?

**Comment by youngkaneda** [comment]
@AlibekJ my English is not so good, but here we go, I care because she's contributed a lot with the community and she just disappeared, why? Some problem with her career, or personal? Is it something we can help with? Is just empathy with the next!

**Comment by leeoniya** [comment]
https://harthur.wordpress.com/2013/01/24/771/

**Comment by Berkmann18** [comment]
> This is one of the most fascinating issues I have ever scrolled through. I love coders and the code community. Way to go for pulling together, people!

Exactly my thought.
I'm surprised in why she didn't set maintainer permissions for those who contacted about that but at least BrainJS is a community thing.

**Comment by dialex** [comment]
Consider this: https://github.com/rejuvenate/rejuvenate
